pkg/services/dashboards/dashboard.go

```diff
@@ -0,0 +1,46 @@
+package dashboards
+
+import (
+	"context"
+
+	"github.com/grafana/grafana/pkg/models"
+)
+
+// DashboardService is a service for operating on dashboards.
+type DashboardService interface {
+	SaveDashboard(ctx context.Context, dto *SaveDashboardDTO, allowUiUpdate bool) (*models.Dashboard, error)
+	ImportDashboard(ctx context.Context, dto *SaveDashboardDTO) (*models.Dashboard, error)
+	DeleteDashboard(ctx context.Context, dashboardId int64, orgId int64) error
+	MakeUserAdmin(ctx context.Context, orgID int64, userID, dashboardID int64, setViewAndEditPermissions bool) error
+	BuildSaveDashboardCommand(ctx context.Context, dto *SaveDashboardDTO, shouldValidateAlerts bool, validateProvisionedDashboard bool) (*models.SaveDashboardCommand, error)
+	UpdateDashboardACL(ctx context.Context, uid int64, items []*models.DashboardAcl) error
+}
+
+//go:generate mockery --name DashboardProvisioningService --structname FakeDashboardProvisioning --inpackage --filename dashboard_provisioning_mock.go
```

Comment:
Not sure we've used the github.com/stretchr/testify/mock a lot earlier. Maybe more in Enterprise. I think we're lacking some instructions/guidance of how this works, how you generate a mock and what to do if you need to update one. Probably as its own sub-section here https://github.com/grafana/grafana/blob/main/contribute/style-guides/backend.md#testing

Can definitely be a separate task after this one is merged.

Nit. Personal taste I would add mocks to a different package, like pkg/services/dashboards/testing. Can be a potential task after this has been merged.

Reply:
I still can't decide on the right subpackage naming scheme here. `testing` would collide with lots of packages, including stdlib.
Naming them `mock` would make everyone who imports them come up with an alias like `dashboardmock` or `mockdashboard` or `dashboardtest` or similar.
Wouldn't it be better to name a package like that from the beginning? Akin to `iotest`, `fstest`, `httptest` and other helpers from the stdlib?